src/Products/Viewer/Cache/FileViewerCache.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading;
+
+namespace GroupDocs.Viewer.MVC.Products.Viewer.Cache
+{
+    internal class FileViewerCache : IViewerCache
+    {
+        private readonly TimeSpan waitTimeout = TimeSpan.FromMilliseconds(100);
+
+        /// <summary>
+        /// Gets the Relative or absolute path to the cache folder.
+        /// </summary>
+        public string CachePath { get; }
+
+        /// <summary>
+        /// Gets the sub-folder to append to the <see cref="CachePath"/>.
+        /// </summary>
+        public string CacheSubFolder { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileViewerCache"/> class.
+        /// </summary>
+        /// <param name="cachePath">Relative or absolute path where document cache will be stored.</param>
+        /// <param name="cacheSubFolder">The sub-folder to append to <paramref name="cachePath"/>.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="cachePath"/> is null.</exception>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="cacheSubFolder"/> is null.</exception>
+        public FileViewerCache(string cachePath, string cacheSubFolder)
+        {
+            if (cachePath == null)
+            {
+                throw new ArgumentNullException(nameof(cachePath));
+            }
+
+            if (cacheSubFolder == null)
+            {
+                throw new ArgumentNullException(nameof(cacheSubFolder));
+            }
+
+            this.CachePath = cachePath;
+            this.CacheSubFolder = cacheSubFolder;
+        }
+
+        /// <summary>
+        /// Serializes data to the local disk.
+        /// </summary>
+        /// <param name="key">An unique identifier for the cache entry.</param>
+        /// <param name="value">The object to serialize.</param>
+        public void Set(string key, object value)
+        {
+            if (value == null)
+            {
+                return;
+            }
+
+            string filePath = this.GetCacheFilePath(key);
+
+            if (value is Stream)
```

Comment:
Codacy found an issue: [Replace this type-check-and-cast sequence with an 'as' and a null check.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933706)